packages/reveal-trigger/src/strategies/lazy-video.js

```diff
@@ -0,0 +1,42 @@
+import invariant from 'tiny-invariant'
+
+class LazyVideoStrategy {
+  constructor(options) {
+    this.options = Object.assign({}, LazyVideoStrategy.defaultOptions, options)
+    this.onNotify = this._onNotify()
+  }
+
+  _onNotify() {
+    return ({ node, entry }) => {
+      if (entry.isIntersecting) {
+        if (node.classList.contains(this.options.previewVideoClass)) {
+          if (node.children.length > 0) {
+            Array.from(node.children).forEach(child => {
+              if (child.tagName === 'SOURCE') {
+                child.src = child.dataset.src
+              }
+            })
+            node.load()
+            node.classList.remove(this.options.previewVideoClass)
+          } else {
+            invariant(false, 'Video node needs at least one source child node.')
```

Comment:
What if the video contains a `src` attribute (Instead of using `<source />`? The check is fine, but I would allow the user to set a single source.

Reply:
> Looks fine to me. What's your thought about animation handling? Should there be an animation mechanism included or not?

I would go without it for the first release, if the feature is needed, it can be added easily at a later point.